nncf/quantization/algorithms/weight_compression/quantize.py

```diff
@@ -0,0 +1,150 @@
+    (scales).
+
+    :param weight: Weight array to compress.
+    :param reduction_axis: Axis, along which to reduce (collect) different statistics (e.g. min, max).
+    :param config: Information on how to compress (quantize) a specific weight.
+    :return: The compressed weights, scale and zero point that was used for its quantization.
+    """
+    mode = config.mode
+    assert mode != CompressWeightsMode.NF4, "The function supports integer quantization only"
+    group_size = config.group_size
+    num_bits = config.num_bits
+
+    level_low = 0
+    level_high = 2**num_bits - 1
+
+    if group_size != -1:
+        # weights are reshaped from [a1, r, a2] to [a1, r//gs, gs, a2]
+        weight, reduction_axis = _reshape_weights_for_grouped_quantization(weight, reduction_axis, group_size)
+
+    if mode in [CompressWeightsMode.INT8_ASYM, CompressWeightsMode.INT4_ASYM]:
+        min_values = np.min(weight, axis=reduction_axis, keepdims=True)  # [a1, r, a2] -> [a1, 1, a2]
+        max_values = np.max(weight, axis=reduction_axis, keepdims=True)  # [a1, r, a2] -> [a1, 1, a2]
+        scale, zero_point = calculate_scale_zero_point(
+            min_values, max_values, level_low, level_high, narrow_range=False
+        )
+    else:
+        scale = np.max(np.abs(weight), axis=reduction_axis, keepdims=True)  # [a1, r//gs, 1, a2]
+        level_low_sym = -(2 ** (num_bits - 1))
+        level_high_sym = 2 ** (num_bits - 1) - 1
+        scale = scale / level_high_sym
+        zero_point = np.array([-level_low_sym])
+
+    eps = np.finfo(weight.dtype).eps
+    # NOTE: adding machine epsilon to avoid division by zero
+    scale[np.abs(scale) < eps] = eps
+    compressed_weights = np.round(weight / scale + zero_point)
+    compressed_weights = np.clip(compressed_weights, level_low, level_high).astype(np.uint8)
+    return compressed_weights, scale, zero_point
+
+
+def _get_integer_quantization_error(weight: np.ndarray, reduction_axis: int, config: WeightCompressionConfig) -> float:
+    """
+    Calculates a quantity characterizing the difference between floating point weights and fake quantized
+    (compressed and decompressed) to integer ones.
+
+    :param weight: Weight array to compress.
+    :param reduction_axis: Axis, along which to reduce (collect) different statistics (e.g. min, max).
+    :param config: Information on how to compress (quantize) a specific weight.
+    :return: The quantity characterizing the error of integer quantization.
+    """
+    orig_shape = weight.shape
+    compressed_weights, scale, zero_point = _do_integer_quantization(weight, reduction_axis, config)
+
+    decompressed_weight = compressed_weights.astype(dtype=scale.dtype)
+    decompressed_weight = (compressed_weights - zero_point) * scale
+
+    decompressed_weight = decompressed_weight.reshape(orig_shape)
+    diff = (decompressed_weight - weight) ** 2
+    layer_err = np.mean(diff, axis=reduction_axis)
+    val = np.max(layer_err)
```

Comment:
That would be super useful for future Torch weight compression to use common experimental tensor over there